dev/sg/sg_deploy.go

```diff
@@ -0,0 +1,251 @@
+  namespace: tooling
+  annotations:
+    kubernetes.io/ingress.class: 'nginx'
+spec:
+  tls:
+    - hosts:
+        - {{.Dns}}
+      secretName: sgdev-tls-secret
+  rules:
+    - host: {{.Dns}}
+      http:
+        paths:
+          - backend:
+              service:
+                name: {{ .Name }}-service
+                port:
+                  number: {{ (index .ServicePorts 0).Port }}
+            path: /
+            pathType: Prefix
+{{- end }}
+`
+
+var dnsTemplate = `
+{{- if .Dns -}}
+# This file was generated by sg deploy.
+
+locals {
+  dogfood_ingress_ip = "34.132.81.184"  # https://github.com/sourcegraph/infrastructure/pull/2125#issuecomment-689637766
+}
+
+resource "cloudflare_record" "{{ .Name }}-sgdev-org" {
+  zone_id = data.cloudflare_zones.sgdev_org.zones[0].id
+  name    = "{{ .Name }}"
+  type    = "A"
+  value   = local.dogfood_ingress_ip
+  proxied = true
+}
+{{- end }}
+`
+
+func generateConfig(configFile string, dryRun bool) error {
+
+	var values Values
+	v, err := os.ReadFile(configFile)
+	if err != nil {
+		return errors.Wrap(err, "read values file")
+	}
+
+	err = yaml.Unmarshal(v, &values)
+	if err != nil {
+		return errors.Wrap(err, "error rendering values")
+	}
+
+	if dryRun {
+
+		fmt.Printf("This is a dry run. The following files would be created:\n\n")
+		t := template.Must(template.New("k8s").Parse(k8sTemplate))
+		err = t.Execute(os.Stdout, &values)
+		if err != nil {
+			return errors.Wrap(err, "execute k8s template")
+		}
+		t = template.Must(template.New("dns").Parse(dnsTemplate))
+		err = t.Execute(os.Stdout, &values)
+		if err != nil {
+			return errors.Wrap(err, "execute dns template")
+		}
+		return nil
+	}
+
+	err = checkCurrentDir("infrastructure")
+	if err != nil {
+		return err
+	}
+
+	k8sPath := fmt.Sprintf("dogfood/kubernetes/tooling/%s", values.Name)
+	err = os.MkdirAll(k8sPath, 0755)
+	if err != nil {
+		return errors.Wrap(err, "create directory")
+	}
+	k8sOutput, err := os.Create(fmt.Sprintf("%s/%s.yaml", k8sPath, values.Name))
+	if err != nil {
+		return errors.Wrap(err, "create file")
+	}
+	defer k8sOutput.Close()
+
+	dnsPath := fmt.Sprintf("dns/%s.sgdev.tf", values.Name)
+	dnsOutput, err := os.Create(dnsPath)
+	if err != nil {
+		return errors.Wrap(err, "create file")
+	}
+	defer dnsOutput.Close()
+	t := template.Must(template.New("k8s").Parse(k8sTemplate))
+	err = t.Execute(k8sOutput, &values)
+	if err != nil {
+		return errors.Wrap(err, "execute k8s template")
+	}
+	t = template.Must(template.New("dns").Parse(dnsTemplate))
+	err = t.Execute(dnsOutput, &values)
+	if err != nil {
+		return errors.Wrap(err, "execute dns template")
+	}
+
+	return nil
+}
+
+func checkCurrentDir(expected string) error {
+
+	cwd, err := os.Getwd()
+	if err != nil {
+		return errors.Wrap(err, "error getting current directory")
+	}
+
+	current := path.Base(cwd)
+	if current != expected {
+		return errors.New("Incorrect directory. Please run from the sourcegraph/infrastructure repository")
```

Comment:
```suggestion
		return errors.New("incorrect directory - please run from the sourcegraph/infrastructure repository")
```
according to gopls errors should not be capatalized